tests/functional/vault/test_withdrawal.py

```diff
@@ -259,3 +259,27 @@ def test_withdrawal_with_reentrancy(
     # given previous setup the withdraw should revert from reentrancy guard
     with brownie.reverts():
         vault.withdraw(vault.balanceOf(gov), {"from": gov})
+
+
+def test_token_transfer_fail(chain, tokenWithBlackList, gov, Vault, TestStrategy):
+    # Need a fresh vault to do this math right
+    vault = Vault.deploy({"from": gov})
+    vault.initialize(
+        tokenWithBlackList,
+        gov,
+        gov,
+        tokenWithBlackList.symbol() + " yVault",
+        "yv" + tokenWithBlackList.symbol(),
+        gov,
+        {"from": gov},
+    )
+    vault.setDepositLimit(2 ** 256 - 1, {"from": gov})
+    starting_balance = tokenWithBlackList.balanceOf(gov)
+
+    tokenWithBlackList.approve(vault, 2 ** 256 - 1, {"from": gov})
+    vault.deposit({"from": gov})
+
+    tokenWithBlackList.blacklistUpdate(gov, True, {"from": gov})
+
+    with brownie.reverts():
+        vault.withdraw(vault.balanceOf(gov), {"from": gov})
```

Comment:
With the change above, this is now what the test can be refactored to

Reply:
I do not see suggested changes.